loadtest/control/simulcontroller/bookmarks.go

```diff
@@ -0,0 +1,153 @@
+		DisplayName: control.PickRandomString(bookmarkNames),
+		Emoji:       emoji,
+		Type:        bookmarkType[rand.Intn(len(bookmarkType))],
+	}
+
+	if bookmark.Type == model.ChannelBookmarkFile {
+		control.AttachFileToBookmark(u, bookmark)
+	} else {
+		bookmark.LinkUrl = control.RandomLink()
+	}
+
+	err = u.AddChannelBookmark(bookmark)
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	return control.UserActionResponse{Info: fmt.Sprintf("bookmark created in channel id %v", channel.Id)}
+}
+
+func (c *SimulController) updateBookmark(u user.User) control.UserActionResponse {
+	if ok, resp := control.ChannelBookmarkEnabled(u); resp.Err != nil {
+		return resp
+	} else if !ok {
+		return control.UserActionResponse{Info: "channel bookmarks not enabled"}
+	}
+
+	channel, err := u.Store().CurrentChannel()
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	currentBookmarks := u.Store().ChannelBookmarks(channel.Id)
+	if len(currentBookmarks) > 0 {
+		// here we update
+		bookmark := currentBookmarks[rand.Intn(len(currentBookmarks))]
+		bookmarkWithFileInfo := bookmark.Clone()
+		bookmarkWithFileInfo.DisplayName = control.PickRandomString(bookmarkNames)
+
+		// 10% of the times bookmarks will have an emoji assigned.
+		if bookmarkWithFileInfo.Emoji == "" && rand.Float64() < 0.1 {
+			bookmarkWithFileInfo.Emoji = control.RandomEmoji()
+		}
+
+		if bookmarkWithFileInfo.Type == model.ChannelBookmarkFile {
+			control.AttachFileToBookmark(u, bookmarkWithFileInfo.ChannelBookmark)
+		} else {
+			bookmarkWithFileInfo.LinkUrl = control.RandomLink()
+		}
+
+		err = u.UpdateChannelBookmark(bookmarkWithFileInfo)
+
+		if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+
+		return control.UserActionResponse{Info: fmt.Sprintf("bookmark %v updated in channel id %v", bookmarkWithFileInfo.Id, channel.Id)}
+	}
+
+	return control.UserActionResponse{Info: fmt.Sprintf("channel id %v does not have bookmarks to update", channel.Id)}
+}
+
+func (c *SimulController) deleteBookmark(u user.User) control.UserActionResponse {
+	if ok, resp := control.ChannelBookmarkEnabled(u); resp.Err != nil {
+		return resp
+	} else if !ok {
+		return control.UserActionResponse{Info: "channel bookmarks not enabled"}
+	}
+
+	channel, err := u.Store().CurrentChannel()
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	currentBookmarks := u.Store().ChannelBookmarks(channel.Id)
+	if len(currentBookmarks) > 0 {
+		bookmark := currentBookmarks[rand.Intn(len(currentBookmarks))]
+		err = u.DeleteChannelBookmark(bookmark.ChannelId, bookmark.Id)
+		if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+
+		return control.UserActionResponse{Info: fmt.Sprintf("bookmark id %v deleted in channel id %v", bookmark.Id, channel.Id)}
+	}
+
+	return control.UserActionResponse{Info: "no channel bookmarks found"}
```

Comment:
Same here. Check first that there are no bookmarks and return early. If there are bookmarks, then proceed with the delete.